### DISCUSSION TITLE
ScriptReference/Behaviour-isActiveAndEnabled

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#200] [Notes]
# ScriptReference/Behaviour-isActiveAndEnabled

https://docs.unity3d.com/ScriptReference/Behaviour-isActiveAndEnabled.html

**Comment by mvi** [comment]
This is the equivalent of:  
  
`behaviour.enabled && behaviour.gameObject.activeInHierarchy`  
It checks that both the behaviour component is enabled as well as the gameObject it lives on being active (and all parent gameObjects recursing upwards).  
Note that like activeInHierarchy, this will return false if used on an uninstantiated prefab (for example if you're writing an editor tool that requires selecting a prefab in the project window).

**Comment by sttz** [comment]
Note that this is **not** exactly equivalent to `enabled && gameObject.activeInHierarchy`.

There was a hint of the difference in the old wording of the documentation (changed in 2021.2):
> Has the Behaviour had active and enabled called?

i.e. `isActiveAndEnabled` does not only check if the behaviour is enabled and the game object active, it additionally checks if the behaviour's `OnEnable` has been called.

So it's possible for both `enabled` and `gameObject.activeInHierarchy` to return `true` but `isActiveAndEnabled` still returning `false`. This can e.g. be the case with multiple scripts on the same game object, inside one script's `OnEnable` when the other script's `OnEnable` has not yet been called, `isActiveEnabled` of the other script will still be `false`.
